core/epochFlags.go

```diff
@@ -0,0 +1,230 @@
+	CheckFrozenCollectionFlag                          EnableEpochFlag = "CheckFrozenCollectionFlag"
+	SendAlwaysFlag                                     EnableEpochFlag = "SendAlwaysFlag"
+	ValueLengthCheckFlag                               EnableEpochFlag = "ValueLengthCheckFlag"
+	CheckTransferFlag                                  EnableEpochFlag = "CheckTransferFlag"
+	TransferToMetaFlag                                 EnableEpochFlag = "TransferToMetaFlag"
+	ESDTNFTImprovementV1Flag                           EnableEpochFlag = "ESDTNFTImprovementV1Flag"
+	ChangeDelegationOwnerFlag                          EnableEpochFlag = "ChangeDelegationOwnerFlag"
+	RefactorPeersMiniBlocksFlag                        EnableEpochFlag = "RefactorPeersMiniBlocksFlag"
+	SCProcessorV2Flag                                  EnableEpochFlag = "SCProcessorV2Flag"
+	FixAsyncCallBackArgsListFlag                       EnableEpochFlag = "FixAsyncCallBackArgsListFlag"
+	FixOldTokenLiquidityFlag                           EnableEpochFlag = "FixOldTokenLiquidityFlag"
+	RuntimeMemStoreLimitFlag                           EnableEpochFlag = "RuntimeMemStoreLimitFlag"
+	RuntimeCodeSizeFixFlag                             EnableEpochFlag = "RuntimeCodeSizeFixFlag"
+	MaxBlockchainHookCountersFlag                      EnableEpochFlag = "MaxBlockchainHookCountersFlag"
+	WipeSingleNFTLiquidityDecreaseFlag                 EnableEpochFlag = "WipeSingleNFTLiquidityDecreaseFlag"
+	AlwaysSaveTokenMetaDataFlag                        EnableEpochFlag = "AlwaysSaveTokenMetaDataFlag"
+	SetGuardianFlag                                    EnableEpochFlag = "SetGuardianFlag"
+	RelayedNonceFixFlag                                EnableEpochFlag = "RelayedNonceFixFlag"
+	ConsistentTokensValuesLengthCheckFlag              EnableEpochFlag = "ConsistentTokensValuesLengthCheckFlag"
+	KeepExecOrderOnCreatedSCRsFlag                     EnableEpochFlag = "KeepExecOrderOnCreatedSCRsFlag"
+	MultiClaimOnDelegationFlag                         EnableEpochFlag = "MultiClaimOnDelegationFlag"
+	ChangeUsernameFlag                                 EnableEpochFlag = "ChangeUsernameFlag"
+	AutoBalanceDataTriesFlag                           EnableEpochFlag = "AutoBalanceDataTriesFlag"
+	FixDelegationChangeOwnerOnAccountFlag              EnableEpochFlag = "FixDelegationChangeOwnerOnAccountFlag"
+	FixOOGReturnCodeFlag                               EnableEpochFlag = "FixOOGReturnCodeFlag"
+	DeterministicSortOnValidatorsInfoFixFlag           EnableEpochFlag = "DeterministicSortOnValidatorsInfoFixFlag"
+)
+
+// CheckHandlerCompatibility checks if the provided handler is compatible with this mx-chain-core-go version
+func CheckHandlerCompatibility(handler EnableEpochsHandler) error {
+	if check.IfNil(handler) {
+		return ErrNilEnableEpochsHandler
+	}
+
+	// allFlags slice must contain all flags defined above
+	allFlags := []EnableEpochFlag{
```

Comment:
For optimization, I think we can have this as an unexported var after the consts definition